api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
 
-@groups.route('/meetings/group/<group_id>')
-@jwt_required
-def read_all_meetings_by_group(group_id):
-    result = db.session.query(Meeting).filter_by(group_id=group_id).all()
+    # 204 codes don't respond with any content
+    return "Deleted successfully", 204
 
-    if len(result) == 0:
-        return jsonify(msg="No meetings found"), 200
 
-    for r in result:
-        r.address = r.address
+# ---- Manager
 
-    return jsonify(meeting_schema.dump(result, many=True))
+manager_schema = ManagerSchema()
 
 
-@groups.route('/meetings/address/<address_id>')
+@groups.route('/groups/<int:group_id>/managers', methods=['POST'])
 @jwt_required
-def read_all_meetings_by_location(address_id):
-    result = db.session.query(Meeting).filter_by(address_id=address_id).all()
+def create_manager(group_id):
+    # make group_id an invalid field in payload
+    manager_schema = ManagerSchema(exclude=['group_id'])
+    try:
+        valid_manager = manager_schema.load(request.json, partial=['active'])
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
-    if len(result) == 0:
-        return jsonify(msg="No meetings found"), 404
+    person_id = valid_manager['person_id']
 
-    return jsonify(meeting_schema.dump(result, many=True))
+    if db.session.query(Manager).filter_by(person_id=person_id, group_id=group_id).first():
+        # if the same manager already exists
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} already exists"), 409
 
+    if 'active' not in valid_manager:
+        valid_manager['active'] = True
 
-@groups.route('/meetings/<meeting_id>')
-@jwt_required
-def read_one_meeting(meeting_id):
-    result = db.session.query(Meeting).filter_by(id=meeting_id).first()
+    new_manager = Manager(group_id=group_id, **valid_manager)
+    db.session.add(new_manager)
+    try:
+        db.session.commit()
+    except IntegrityError:
+        return jsonify('Payload contains an invalid foreign key'), 404
+    return jsonify(manager_schema.dump(new_manager)), 201
 
-    if result is None:
-        return jsonify(msg="Meeting not found"), 404
 
-    result.address = result.address
+@groups.route('/groups/<int:group_id>/managers')
+@jwt_required
+def read_all_managers(group_id):
+    query = db.session.query(Manager).filter_by(group_id=group_id)
+    try:
+        managers = get_all_queried_entities(query, request.args)
+    except QueryArgumentError as e:
+        return jsonify(e.message), e.code
+    return jsonify(manager_schema.dump(managers, many=True))
 
-    return jsonify(meeting_schema.dump(result))
 
+@groups.route('/groups/<int:group_id>/managers/<int:person_id>')
+@jwt_required
+def read_one_manager(group_id, person_id):
+    manager = db.session.query(Manager).filter_by(group_id=group_id, person_id=person_id).first()
+    if manager is None:
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} does not exist"), 404
```

Comment:
Similarly confusing error message. Please create an issue that asks for all error messages in CC to be reviewed to try to make them as helpful to the end user as possible.

Reply:
Will do that. But just a thought: if the the APIs are primarily to be consumed by the UI, should the UI be responsible for making a request to a valid entity? Do we intend to display the error from the API directly in the user interface rather than transforming it with some extra logic? If not, I think error messages like this can be more helpful in debugging, even though it does not make any sense for an end user.

Reply:
We want both useful UI and info for debugging. I think this is covered by the plan to improve error reporting to the end user and adding a logging mechanism on the server side.